internal/pkg/agent/transpiler/ast.go

```diff
@@ -58,6 +59,9 @@ type Node interface {
 	// Hash compute a sha256 hash of the current node and recursively call any children.
 	Hash() []byte
 
+	// Hash64With recursively computes the given hash for the Node and its children
+	Hash64With(h *xxhash.Digest) error
```

Comment:
Why not just replace the usage of `Hash()`? Why do you have both? Do we need both?

Reply:
I don't think we need both. If there were a bunch of callers of `Hash`, then I'd be in favor of keeping it and implementing it using `Hash64With`, but there aren't, so we can get rid of it. I was planning to do so in a follow up.